pyworkflow/pyworkflow/workflow.py

```diff
@@ -326,79 +319,57 @@ def retrieve_node_data(node_to_retrieve):
             raise WorkflowException('retrieve node data', str(e))
 
     @staticmethod
-    def read_graph_json(file_like):
+    def read_graph_json(json_data):
         """Deserialize JSON NetworkX graph
 
         Args:
-            file_like: file-like object from which to read JSON-serialized graph
+            json_data: JSON data from which to read JSON-serialized graph
 
         Returns:
              NetworkX DiGraph object
 
         Raises:
-            OSError: on file error
             NetworkXError: on issue with loading JSON graph data
         """
-        json_data = json.load(file_like)
         return nx.readwrite.json_graph.node_link_graph(json_data)
 
     @staticmethod
     def generate_file_name(workflow, node_id):
```

Comment:
The method is only used once, in `store_node_data()`, and I thought about removing it completely. Once generated and used for writing data, they can be looked up by accessing the `data` attribute on a Node, which I think is all we need (basically your idea of an instance method on the `Node` class?).

If there is a reason to keep this, I think it makes more sense in the `Workflow` class since the file name includes the Workflow name.